client/src/page-not-found/index.tsx

```diff
@@ -31,7 +31,7 @@ export function NoMatch() {
   // All of this should be done in a lazy-loaded module.
 
   return (
-    <div className="no-match">
+    <div className="page-not-found">
```

Comment:
Do we need this `div`? From what I saw in devtools, everything works as expected if you only keep the `main` element below.

```
<main className="page-content-container page-not-found" role="main">
```

Reply:
No doubt do we need to add the className `page-not-found` anyway so you can connect to this DOM in `page-not-found/index.scss`. 
The question is; What is the correct canonical wrapper we should be using? 
The `<Document>` is the first and only real page we have (apart from unimportant non-production pages like the writer's home page, the Flaws dashboard, etc). 
The document has the following structure:
```html
<div class="page-wrapper reference-page">
  ...TITLE...
  ...BREADCRUMB STUFF...
  <div class="page-content-container">
    ...TOC...
    <main id="content" class="main-content" role="main">
      <article class="article">
          <RenderDocumentBody doc={doc} />
```

(it's pretty intense that it needs 4 wrapper blocking elements but I'm sure there's a good reason for I don't understand)

So what should be the correct "pattern" for ALL other pages that are NOT a document? E.g. 404, home page, site-search, sign-up, user profiles, etc. 

The reason I ask is because I'd like to take what we decide and extract it into a molecule (or atom) so that we don't have to type variations of this in more than one place. Now that we have more than exactly 1 type of page, it would make sense to write (wrapper) components like `<PageContentContainer>` or `<MainContentContainer>`. 

Once that's settled, we can decide whether we want to have:
```diff
<main id="content" class="main-content" role="main">
+  <div class="page-not-found">
```
...or...
```diff
-<main id="content" class="main-content" role="main">
+<main id="content" class="main-content page-not-found" role="main">
```
...for example.

Reply:
Ah yes so, for the reference(document) page type we need that pattern because of the way the various pages of the main content area flows on different viewports. For this `standard` page type, the layout is much simpler and so the DOM structure is also much simpler.

I would therefore say for this, the patter will be:

```
<main className="page-content-container page-not-found" role="main">
```

The `page-not-found` specific class is there in case we want to add some custom styling that is only to be applied to the 404 page for example. Other than that, it is not strictly needed. i.e. the following is the minimum we need for the basic layout:

```
<main className="page-content-container" role="main">
```

Reply:
What about the `<div class="page-content-container">`?

Reply:
For `standard` we should use the `main` element because all the main content will live directly inside it. Unlike the `reference/document` page which has a combination of the toc, sidebar, and main content inside the `page-content-container`